client/components/sub-masterbar-nav/index.jsx

```diff
@@ -0,0 +1,161 @@
+			} )
+		),
+		uri: PropTypes.string.isRequired
+	};
+
+	static defaultProps = {
+		options: []
+	}
+
+	state = {
+		collapsed: true
+	};
+
+	componentDidMount() {
+		this.onResize();
+
+		window.addEventListener( 'resize', this.onResize.bind( this ) );
+		window.addEventListener( 'scroll', this.onScroll.bind( this ) );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.onResize.bind( this ) );
+		window.removeEventListener( 'scroll', this.onScroll.bind( this ) );
+	}
+
+	render() {
+		const className = classNames(
+			'sub-masterbar-nav',
+			{ 'is-collapsed': this.state.collapsed }
+		);
+
+		const ellipsisClass = classNames(
+			'sub-masterbar-nav__ellipsis',
+			{ 'is-open': ! this.state.collapsed }
+		);
+
+		return (
+			<div className={ className }>
+				{ this.isDroppable( this.state.width ) &&
+					<div className="sub-masterbar-nav__select" onClick={ this.toggleList.bind( this ) }>
+						<Item
+							isSelected={ true }
+							label={ this.getSelected().label }
+							icon={ this.getSelected().icon }
+						/>
+						<Gridicon icon="chevron-down" className="sub-masterbar-nav__select-icon" />
+					</div>
+				}
+				<div className="sub-masterbar-nav__wrapper">
+					<div className="sub-masterbar-nav__items">
+						{ this.renderItems() }
+					</div>
+				</div>
+				{ this.isFoldable( this.state.width, this.props.options ) && (
+					<div className="sub-masterbar-nav__switch">
+						<Gridicon icon="ellipsis" className={ ellipsisClass } onClick={ this.toggleList.bind( this ) } />
+					</div>
+				)}
+			</div>
+		);
+	}
+
+	renderItems() {
+		return this.props.options.map( ( item, index ) =>
+			<Item
+				key={ index }
+				isSelected={ this.isSelected( item ) }
+				onClick={ this.selectItem.bind( this, item ) }
+				label={ item.label }
+				icon={ item.icon }
+				href={ item.uri }
+			/>
+		);
+	}
+
+	selectItem( option ) {
+		if ( ! option ) {
+			return;
+		}
+
+		this.setState( ( state ) => ( {
+			collapsed: this.isDroppable( state.width ) || state.collapsed
+		} ) );
+	}
+
+	toggleList() {
+		this.setState( ( state ) => ( {
+			collapsed: ! state.collapsed
+		} ) );
+	}
+
+	onScroll() {
+		this.setState( ( state ) => ( {
+			collapsed: state.collapsed || this.isDroppable( state.width )
+		} ) );
+	}
+
+	onResize() {
+		const currentWidth = findDOMNode( this ).offsetWidth;
+
+		this.setState( ( state, props ) => ( {
+			width: currentWidth,
+			collapsed: state.collapsed || this.hasChangedView( state.width, currentWidth, props.options )
+		} ) );
+	}
+
+	hasChangedView( oldWidth, newWidth, options ) {
+		const min = Math.min( oldWidth, newWidth );
+		const max = Math.max( oldWidth, newWidth );
+
+		return ( this.isDroppable( min ) && ! this.isDroppable( max ) ) ||
+			( this.isFoldable( min, options ) && ! this.isFoldable( max, options ) );
+	}
+
+	isFoldable( width, options ) {
+		return VIEW_BREAKPOINT < width && width < options.length * ITEM_WIDTH + VIEW_PADDING;
```

Comment:
I'm not actually sure it would as it pulls the width from the window instead of taking a parameter.  
This would cause problems for ```hasChangedView``` which collapses the dropdown/list whenever the breakpoint is crossed.

If we don't care about that, I can remove that function and use ```isWithinBreakpoint``` where appropriate.